### PR TITLE
[WIPTEST]Added safer way of handling discovered machines with nicer output

### PR DESCRIPTION
**Pull request by jistefan**
… in case of failure

Purpose or Intent
=================

__Extending__ provider discovery because previous approach was not safe enough. New one has nice output in case of failure.

{{pytest: -k 'test_discover_infra' --use-provider complete -s -v}}

### CODE REVIEW

**Comment by jistefan** [comment]
There is problem with specific provider, that will be solved in separate PR.

File: cfme/tests/infrastructure/test_provider_discovery.py [hunk 1]
```diff
@@ -103,14 +105,18 @@ def test_discover_infra(providers_for_discover, start_ip, max_range):
 
     discover(rhevm, virtualcenter, scvmm, False, start_ip, max_range)
 
-    @pytest.wait_for(num_sec=count_timeout(start_ip, max_range), delay=5)
-    def _wait_for_all_providers():
-        for provider in providers_for_discover:
-            if provider.key not in pytest.store.current_appliance.managed_providers:
+    try:
+        @pytest.wait_for(num_sec=count_timeout(start_ip, max_range), delay=5)
```
Comment (RonnyPfannschmidt) [inline cfme/tests/infrastructure/test_provider_discovery.py]: @psav we should deprecate the pytest namespace usage in wait_for

Comment (psav) [inline cfme/tests/infrastructure/test_provider_discovery.py]: Absolutely

Comment (jistefan) [inline cfme/tests/infrastructure/test_provider_discovery.py]: @psav @RonnyPfannschmidt May I ask You for example of valid code, please?

Comment (mfalesni) [inline cfme/tests/infrastructure/test_provider_discovery.py]: @jistefan `utils.wait.wait_for_decorator` ... wait, we don't wrap that one.

```python
# utils/wait.py
# modify
from wait_for import wait_for as wait_for_mod, wait_for_decorator as wait_for_decorator_mod

# add at the end
wait_for_decorator = partial(wait_for_decorator_mod, logger=logger)
```

Then in your file:

```python
from utils.wait import wait_for_decorator

@wait_for_decorator(num_sec=count_timeout(start_ip, max_range), delay=5)
def foo():
    bar()
```

**Comment by dajoRH** [comment]
Lint report for commit 88067c2d6f0ce06426cc93b0505ad2c628104385:

Lint OK! :cake: :punch: :cookie: .

* 7692b13267c677090239188a8a899fe0b9e20008
  * Merge commit detected!!! :bangbang:

**The PR might not be acceptable in this state** :rage4:.
There is a merge commit in the branch which we do not like in PRs so please, get rid of it.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 61221641248f99cb94e2beba68a450f1275072aa:

`utils/wait.py`:
- :red_circle: Line 5:1: **F811** *redefinition of unused 'wait_for_mod' from line 1*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

File: cfme/tests/infrastructure/test_provider_discovery.py [hunk 2]
```diff
@@ -103,14 +105,18 @@ def test_discover_infra(providers_for_discover, start_ip, max_range):
...
-    def _wait_for_all_providers():
-        for provider in providers_for_discover:
-            if provider.key not in pytest.store.current_appliance.managed_providers:
+    try:
+        @wait_for_decorator(num_sec=count_timeout(start_ip, max_range), delay=5)
+        def _wait_for_all_providers():
+            for provider in providers_for_discover:
+                if provider.key not in pytest.store.current_appliance.managed_providers:
+                    return False
+            if len(pytest.store.current_appliance.managed_providers) != len(providers_for_discover):
```
Comment (psav) [inline cfme/tests/infrastructure/test_provider_discovery.py]: Though we should never have more than X providers, I think we need to either change the fixture above to ensure there are NO providers, or drop the check for the number of providers. Why did we add this? Is checking for one list against the other not enough?

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Pull request closed without merging**